flocker/node/agents/ebs.py

```diff
@@ -160,7 +282,7 @@ def create_volume(self, dataset_id, size):
         Open issues: https://clusterhq.atlassian.net/browse/FLOC-1792
         """
         requested_volume = self.connection.create_volume(
-            size=int(Byte(size).to_GB().value), zone=self.zone)
+            size=int(Byte(size).to_GiB().value), zone=self.zone)
```

Comment:
Truncating here will under-allocate volumes and probably also cause the convergence algorithm to never converge.  I think the same is true of the OpenStack implementation of this method.  It doesn't explicitly truncate to int but I strongly suspect the Cinder library we're using does it for us (because the Cinder HTTP API documentation says size must be an integer).

So in both cases if someone asks for a number of bytes that doesn't divide evenly into the unit supported by the backend, they'll get a slightly too-small volume and the convergence agent will then infinitely loop trying to resize it to be the right size (which of course will fail since no one has implemented resize yet :smile: ).

FLOC-1874 seems particularly relevant here.  So maybe just add a link to it here.